File: lessonlink-react-app/src/dtos/RegisterTeacherDto.ts
export type RegisterTeacherDto = {
    firstName: string;
    surName: string;
    email: string;
    password: string;
    acceptsOnline: boolean;
    acceptsInPerson: boolean;
    location: string | null;
    hourlyRate: number;
    contact: string;
    subjectNames: string[];
};

File: LessonLink/LessonLink.Domain/DTOs/User/RegisterTeacherDto.cs
using System.ComponentModel.DataAnnotations;

namespace LessonLink.BusinessLogic.DTOs.User;

public class RegisterTeacherDto
{
    public required string FirstName { get; set; }
    public required string SurName { get; set; }

    [EmailAddress]
    public required string Email { get; set; }
    public required string Password { get; set; }

    public required bool AcceptsOnline { get; set; }
    public required bool AcceptsInPerson { get; set; }
    public required string? Location { get; set; }
    public required int HourlyRate { get; set; }
    public required string Contact { get; set; }
    public required List<string> SubjectNames { get; set; }
}
